### ISSUE TITLE
False positives for exception expression

### ISSUE THREAD

**Issue by Chi-teck** [squizlabs/PHP_CodeSniffer#3919]
## Describe the bug
Since PHP 8.0 expressions can now throw exceptions.
So the code below is valid. However, it triggers `Squiz.PHP.NonExecutableCode.Unreachable` warning.

### Code sample
```php
$input = TRUE;
$input ?: throw new \RuntimeException('Error.');
echo 'The code was executed';
```

**Comment by jrfnl** [comment]
Closing as already fixed via #3777, which will be included in PHPCS 3.8.0.

**Comment by jrfnl** [comment]
FYI: the fix for this issue is included in today's [PHP_CodeSniffer 3.8.0 release](https://github.com/PHPCSStandards/PHP_CodeSniffer/releases/tag/3.8.0).

As per #3932, development on PHP_CodeSniffer will continue in the [PHPCSStandards/PHP_CodeSniffer](https://github.com/PHPCSStandards/PHP_CodeSniffer) repository. If you want to stay informed, you may want to start "watching" that repo (or watching releases from that repo).
